src/panels/lovelace/components/hui-marquee.ts

```diff
@@ -0,0 +1,88 @@
+import {
+  html,
+  LitElement,
+  PropertyValues,
+  TemplateResult,
+  customElement,
+  css,
+  CSSResult,
+  property,
+} from "lit-element";
+
+const marqueeSpeed = 0.2;
+
+@customElement("hui-marquee")
+class HuiMarquee extends LitElement {
+  @property() public text?: string;
+  @property() public active?: boolean;
+  private _interval?: number;
+  private _left: number = 0;
+
+  protected updated(changedProperties: PropertyValues): void {
+    super.updated(changedProperties);
+
+    if (
+      changedProperties.has("active") &&
+      this.active &&
+      !this._interval &&
+      this.offsetWidth < this.scrollWidth
+    ) {
+      this._interval = window.setInterval(() => {
+        this._play();
+      });
+
+      this.requestUpdate();
+    }
+  }
+
+  protected render(): TemplateResult {
+    if (!this.text) {
+      return html``;
+    }
+
+    return html`
+      <div>${this.text}</div>
+      ${this._interval
+        ? html`
+            <div>${this.text}</div>
+          `
+        : ""}
+    `;
+  }
+
+  private get _marqueeElementFirstChild(): HTMLElement {
+    return this.shadowRoot!.firstElementChild as HTMLElement;
+  }
+
+  private _play(): void {
+    this.style.marginLeft = "-" + this._left + "px";
+
+    if (!this.active && !this._left) {
+      clearInterval(this._interval);
+      this._interval = undefined;
+    }
```

Comment:
Should there be a return here? And should we reset `this.style.marginLeft`?

Reply:
Should be a return. But we already reset the margin in the beginning of the play function

Reply:
We set the marginLeft, but if we stop because `active` is `false`, `left` is not 0 while we stop, we won't go back to the original location ?

Reply:
? The if statement checks and only stops when left is 0

Reply:
Its waits until the Marquee returns to its original position before stopping. Therefore, this.left is 0, margin is 0 and active is false

Reply:
oh my bad, sorry. I somehow thought it was an `||`